src/app/zap-templates/zcl/data-model/chip/scene.xml

```diff
@@ -0,0 +1,235 @@
+   
+    <command source="client" code="0x06" name="GetSceneMembership" response="GetSceneMembershipResponse" optional="false" cli="chip scenes get">
+      <description>
+        Get an unused scene identifier when no commissioning tool is in the network, or for a commissioning tool to get the used scene identifiers within a certain group 
+      </description>
+      <arg name="groupId" type="group_id"/>
+    </command>
+
+    <command source="client" code="0x40" name="EnhancedAddScene" response="EnhancedAddSceneResponse" optional="true" cli="chip scenes eadd">
+      <description>
+        Allows a scene to be added using a finer scene transition time than the AddScene command.
+      </description>
+      <arg name="groupId" type="group_id"/>
+      <arg name="sceneId" type="INT8U"/>
+      <arg name="transitionTime" type="INT16U"/>
+      <arg name="sceneName" type="CHAR_STRING"/>
+      <arg name="extensionFieldSets" type="SceneExtensionFieldSet" array="true"/>
+    </command>
+    <command source="client" code="0x41" name="EnhancedViewScene" response="EnhancedViewSceneResponse" optional="true" noDefaultImplementation="true" cli="chip scenes eview">
+      <description>
+        Allows a scene to be retrieved using a finer scene transition time than the ViewScene command
+      </description>
+      <arg name="groupId" type="group_id"/>
+      <arg name="sceneId" type="INT8U"/>
+    </command>
+    <command source="client" code="0x42" name="CopyScene" response="CopySceneResponse" optional="true" cli="chip scenes copy">
+      <description>
+        Allows a client to efficiently copy scenes from one group/scene identifier pair to another group/scene identifier pair.
+      </description>
+      <arg name="mode" type="ScenesCopyMode"/>
+      <arg name="groupIdFrom" type="group_id"/>
+      <arg name="sceneIdFrom" type="INT8U"/>
+      <arg name="groupIdTo" type="group_id"/>
+      <arg name="sceneIdTo" type="INT8U"/>
+    </command>
+    
+    <command source="server" code="0x00" name="AddSceneResponse" optional="false" disableDefaultResponse="true">
+      <description>
+        The command is generated in response to a received unicast AddScene command,
+      </description>
+      <arg name="status" type="ENUM8"/>
+      <arg name="groupId" type="group_id"/>
+      <arg name="sceneId" type="INT8U"/>
+    </command>
+    
+    <command source="server" code="0x01" name="ViewSceneResponse" optional="false" disableDefaultResponse="true">
+      <description>
+        The command is generated in response to a received unicast ViewScene command 
+      </description>
+      <arg name="status" type="ENUM8"/>
+      <arg name="groupId" type="group_id"/>
+      <arg name="sceneId" type="INT8U"/>
+      <arg name="transitionTime" type="INT16U" presentIf="status==0"/>
+      <arg name="sceneName" type="CHAR_STRING" presentIf="status==0"/>
+      <arg name="extensionFieldSets" type="SceneExtensionFieldSet" array="true" presentIf="status==0"/>
```

Comment:
The spec is self-contradictory here: it claims the fields are mandatory, but then omits them in some cases.  Please raise a SDK feedback spec issue to mark the fields optional?   This needs to get resolved in the next several days...

And in SDK implementation terms presentIf does absolutely nothing; if we want the fields to be omittable, they need to be optional.

Reply:
I agree, I raised your point in the matter-scene discussion channel. I also replaced the presentIf by optional=true